sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample13_AdvancedConfiguration.md

```diff
@@ -74,3 +74,122 @@ ServiceBusProcessor processor = client.CreateProcessor("<queue-name>", new Servi
     PrefetchCount = 10
 });
 ```
+
+## Configuring a lock lost handler when using the processor
+
+The processor allows for automatic lock renewal based on the configured 
+`MaxAutoLockRenewalDuration` property. However, there are times when the lock may be lost due to 
+a transient network issue. The lock can also be considered lost if the processing time exceeds 
+the `MaxAutoLockRenewalDuration`. For these cases, the processor allows you to register an event 
+handler for an event that is raised when the lock is lost. In the below examples, we show how 
+the overall processing token can be linked to the lock lost event handler so that you can avoid 
+expensive processing when you know the processor is shutting down or the message/session lock 
+has been lost.
+
+When using the `ServiceBusProcessor`:
+
+```C# Snippet:ServiceBusProcessorLockLostHandler
+string connectionString = "<connection_string>";
+var client = new ServiceBusClient(connectionString);
+
+// create a processor that we can use to process the messages
+await using ServiceBusProcessor processor = client.CreateProcessor("<queue-name>");
+
+// configure the message and error handler to use
+processor.ProcessMessageAsync += MessageHandler;
+processor.ProcessErrorAsync += ErrorHandler;
+
+async Task MessageHandler(ProcessMessageEventArgs args)
+{
+    using var cts = CancellationTokenSource.CreateLinkedTokenSource(args.CancellationToken);
+    args.MessageLockLostAsync += MessageLockLostHandler;
```

Comment:
We probably should have a `try/finally` that unwires the lock lost handler so that it happens regardless of exceptions.